lib/bolt/pal.rb

```diff
@@ -63,13 +69,21 @@ def initialize(modulepath, hiera_config, resource_types, max_compiles = Etc.npro
       @project = project
 
       @logger = Bolt::Logger.logger(self)
-      if modulepath && !modulepath.empty?
-        @logger.debug("Loading modules from #{@modulepath.join(File::PATH_SEPARATOR)}")
+      unless @modulepath.apply_compiler_modulepath.empty?
```

Comment:
Yeah, I think this needs a rename. I think the bolt_compiler_modulepath makes sense as a name, since it is actually that: the modulepath that the bolt compiler needs. But I think this other modulepath needs another name.